f8a_worker/workers/sentiment_scorer.py

```diff
@@ -0,0 +1,324 @@
+                   auth_uri=auth_uri,
+                   token_uri=token_uri,
+                   auth_provider_cert_url=auth_provider_cert_url,
+                   client_url=client_url)
+
+    key_file = open(local_path, "w")
+    key_file.write(key_file_contents)
+    key_file.close()
+
+
+def sentiment_analysis(arg):
+    key_file = "/tmp/key.json"
+    write_key_file(local_path=key_file)
+    ecosystem, pkg_name = arg
+
+    # First, check if package sentiment is available in cache i.e. graph db
+    sentiment_cache = SentimentGraphStore(graph_db_url=GREMLIN_SERVER_URL_REST)
+    sentiment_output = sentiment_cache.get_sentiment_details(key_name=pkg_name)
+    if len(sentiment_output.items()) == 0:  # package not found in graph i.e. unknown
+        return {}
+
+    # Let us now check if package sentiment is fresh enough or not
+    flag_refresh = True
+    today_date = datetime.datetime.today()
+    if sentiment_output['last_updated']:
+        last_updated = datetime.datetime.strptime(sentiment_output['last_updated'],
+                                                  "%Y-%m-%d")
+        time_diff = today_date - last_updated
+        if time_diff.days <= 7:  # Found data in cache
+            flag_refresh = False
+
+    # Compute fresh package-sentiment if required
+    if flag_refresh:
+        # TODO decide stack-overflow tag for other ecosystems
+        search_tag = 'python' if ecosystem == 'pypi' else ecosystem
+        data_store = GooglePublicDataStore(key_file_path=key_file)
+        text = data_store.get_stack_overflow_data(search_keyword=pkg_name,
+                                                  search_tag=search_tag,
+                                                  num_months=6)
+
+        sentiment_analyzer = GoogleSentimentAnalyzer(key_file_path=key_file)
+        score, magnitude = sentiment_analyzer.analyze_sentiment(text=text)
+
+        sentiment_output = {
+            'package_name': pkg_name,
+            'score': score,
+            'magnitude': magnitude,
+            'last_updated': today_date.strftime("%Y-%m-%d")
+        }
+
+        sentiment_cache.set_sentiment_details(key_name=pkg_name,
+                                              key_value=sentiment_output)
+
+    return sentiment_output
+
+
+class UserStackSentimentScoringTask(BaseTask):
+    description = 'Sentiment scoring for the packages in user-stack'
+    _analysis_name = 'user_stack_sentiment_scorer'
+
+    def execute(self, arguments=None):
+        aggregated = self.parent_task_result('GraphAggregatorTask')
+
+        sentiment_output = {}
+        arg_instances = []
+        for result in aggregated['result']:
+            resolved = result['details'][0]['_resolved']
+            ecosystem = result['details'][0]['ecosystem']
+            for elm in resolved:
+                arg_instances.append((ecosystem, elm['package']))
+
+        sentiment_results = Parallel(n_jobs=4, verbose=1, backend="threading")\
```

Comment:
I think this won't work as expected, but I can be wrong. In any case I am not going to block PR on this as this is just an optimization.

https://pythonhosted.org/joblib/parallel.html#using-the-threading-backend


> If you know that the function you are calling is based on a compiled extension that releases the Python Global Interpreter Lock (GIL) during most of its computation then it might be more efficient to use threads instead of Python processes as concurrent workers. For instance this is the case if you write the CPU intensive part of your code inside a with nogil block of a Cython function.

`sentiment_analysis()` will, in my opinion, hold the GIL and won't release it.